gnmi_server/server.go

```diff
@@ -21,15 +21,33 @@ import (
 	"google.golang.org/grpc/peer"
 	"google.golang.org/grpc/reflection"
 	"google.golang.org/grpc/status"
+	"google.golang.org/grpc/keepalive"
 	"net"
 	"strings"
 	"sync"
+	"time"
 )
 
 var (
 	supportedEncodings = []gnmipb.Encoding{gnmipb.Encoding_JSON, gnmipb.Encoding_JSON_IETF}
 )
 
+var (
+	keep_alive_enforcement = keepalive.EnforcementPolicy{
+		MinTime:            5 * time.Second,
+		PermitWithoutStream: true,
+	}
+)
+
+var (
+	keep_alive_params = keepalive.ServerParameters{
```

Comment:
>ServerParameters [](http://example.com/codeflow?start=31&length=16)

Does any parameter impact on-demand subscription? For example if subscribing a very stable counter, and there is no counter change for days, will it be disconnected?

Reply:
No, it will not be disconnected. This change only affects closing idle connections, which means ON_CHANGE connections won't be disconnected even if there are no updates. Tested ON_CHANGE with "STATE_DB/DEVICE_METADATA" where no updates were happening. Connection did not close.